packages/gafl-webapp-service/src/pages/contact/digital-licence/licence-confirmation-method/__tests__/result-function.spec.js

```diff
@@ -0,0 +1,36 @@
+import resultFunction from '../result-function'
+import { CommonResults } from '../../../../../constants.js'
+
+describe('newsletter > result-function', () => {
+  const getMockRequest = (fromContactDetailsSeen, summary) => ({
+    cache: () => ({
+      helpers: {
+        status: {
+          getCurrentPermission: async () => ({
+            fromContactDetailsSeen: fromContactDetailsSeen,
+            fromSummary: summary
+          })
+        }
+      }
+    })
+  })
+
+  describe('result function', () => {
+    beforeEach(jest.clearAllMocks)
+
+    it('should return amend if status.fromContactDetailsSeen equals seen', async () => {
+      const result = await resultFunction(getMockRequest('seen'))
+      expect(result).toBe(CommonResults.AMEND)
+    })
+
+    it('should return summary if status.fromContactDetailsSeen does not equal seen and status.fromSummary equals contact-summary', async () => {
+      const result = await resultFunction(getMockRequest('not-seen', 'contact-summary'))
+      expect(result).toBe(CommonResults.SUMMARY)
+    })
+
+    it('should return ok if status.fromContactDetailsSeen does not equal seen and status.fromSummary does not equal contact-summary', async () => {
```

Comment:
you're omitting that it can't be a renewal either - a better description might be 'should return ok if contact details haven't been seen, it's returning from the contact summary and it's not a renewal'. Think of why something is happening: 'x does not equal y' is an ok description, but 'such and such a condition hasn't happened' is better as it's plain English and describes in human terms what's happening.